python/lsst/display/firefly/firefly.py

```diff
@@ -185,17 +187,25 @@ def _mtv(self, image, mask=None, wcs=None, title=""):
                 fdm.seek(0,0)
                 self._fireflyMaskOnServer = _fireflyClient.upload_data(fdm, 'FITS')
 
-            self._maskDict = mask.getMaskPlaneDict()
+            maskPlaneDict = mask.getMaskPlaneDict()
+            for k, v in maskPlaneDict.items():
+                self._maskDict[k] = v
+                self._maskPlaneColors[k] = self.display.getMaskPlaneColor(k)
             usedPlanes = long(afwMath.makeStatistics(mask, afwMath.SUM).getValue())
             for k in self._maskDict:
-                if ((1 << self._maskDict[k]) & usedPlanes):
+                if (((1 << self._maskDict[k]) & usedPlanes) and
+                    (k in self._maskPlaneColors) and
```

Comment:
Can this condition `(k in self._maskPlaneColors)` ever be false?

Reply:
Good question! Looking at an example, `_maskPlaneColors` has fewer keys than `_maskDict` (which seems to have all possibilities) so it's possible to reference a key not in `_maskPlaneColors`.

Reply:
Looking at line 192 and 193, I thought `_maskDict` and `_maskPlaneColors` would have identical keys. But I didn't read other part of the code so probably missed some subtleties.